packages/ajax/test/AjaxClass.interceptors-cache.test.js

```diff
@@ -0,0 +1,713 @@
+        expect(server.requests.length).to.equal(1);
+      })
+      .then(() => {
+        clock.tick(1000 * 60 * 59); // 0:59 hour
+      })
+      .then(() => ajax.get('/test-hour'))
+      .then(() => {
+        expect(server.requests.length).to.equal(1);
+        clock.tick(1000 * 60 * 61); // 1:01 hour
+      })
+      .then(() => ajax.get('/test-hour'))
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .finally(() => {
+        ajaxGetSpy.restore();
+        clock.restore();
+        removeCacheInterceptors(ajax, indexes);
+      });
+  });
+
+  it('invalidates invalidateUrls endpoints', () => {
+    newCacheId();
+
+    const indexes = addCacheInterceptors(ajax, {
+      useCache: 'always',
+      timeToLive: 500,
+    });
+    const ajaxGetSpy = sinon.spy(ajax, 'get');
+    const ajaxPostSpy = sinon.spy(ajax, 'post');
+
+    server.respondWith(/\/test-valid-url/, xhr => {
+      xhr.respond(
+        200,
+        { 'Content-Type': 'application/json' },
+        JSON.stringify({ some: 'data-valid-url' }),
+      );
+    });
+
+    server.respondWith(/\/test-invalid-url/, xhr => {
+      xhr.respond(
+        200,
+        { 'Content-Type': 'application/json' },
+        JSON.stringify({ some: 'data-invalid-url' }),
+      );
+    });
+    const actionConfig = {
+      lionCacheOptions: {
+        invalidateUrls: ['/test-invalid-url'],
+      },
+    };
+
+    return ajax
+      .get('/test-valid-url', actionConfig)
+      .then(() => {
+        expect(ajaxGetSpy.calledOnce).to.be.true;
+        expect(ajaxGetSpy.calledWith('/test-valid-url')).to.be.true;
+        expect(server.requests.length).to.equal(1);
+      })
+      .then(() => ajax.get('/test-invalid-url'))
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .then(() => ajax.post('/test-valid-url', {}, actionConfig))
+      .then(() => {
+        expect(ajaxPostSpy.calledOnce).to.be.true;
+        expect(ajaxPostSpy.calledWith('/test-valid-url')).to.be.true;
+        expect(server.requests.length).to.equal(3);
+      })
+      .then(() => ajax.get('/test-invalid-url'))
+      .then(() => {
+        expect(server.requests.length).to.equal(4);
+      })
+      .finally(() => {
+        ajaxGetSpy.restore();
+        ajaxPostSpy.restore();
+        removeCacheInterceptors(ajax, indexes);
+      });
+  });
+
+  it('throws error when invalidateUrls is not an array', () => {
+    newCacheId();
+
+    const indexes = addCacheInterceptors(ajax, {
+      useCache: 'always',
+    });
+    expect(() => {
+      ajax.get('/test', {
+        lionCacheOptions: validateOptions({
+          // @ts-ignore needed for test
+          invalidateUrls: 'not an array',
+        }),
+      });
+    }).to.throw(/Property `invalidateUrls` must be of type `Array` or `falsy`/);
+    removeCacheInterceptors(ajax, indexes);
+  });
+
+  it('throws error when invalidateUrls is applied to global config', () => {
```

Comment:
Is there a specific reason why invalidateUrls / invalidateRegex can only be applied per action to force a "real" response?

Why not allow to set it on the global config too, e.g. because we never want cached response for certain urls? Then the user doesn't have to pass it as an action config every time the call it

Reply:
The intention was allow invalidation of the get responses which are owned by the user, but in fact you can invalidate other responses if you want. Since we cannot limit it anyhow probably better idea just to allow it from any config